chain/sub/incoming.go

```diff
@@ -85,9 +165,9 @@ func HandleIncomingMessages(ctx context.Context, mpool *messagepool.MessagePool,
 			continue
 		}
 
-		m, err := types.DecodeSignedMessage(msg.GetData())
-		if err != nil {
-			log.Errorf("got incorrectly formatted Message: %s", err)
+		m, ok := msg.ValidatorData.(*types.SignedMessage)
```

Comment:
if the first step after receiving a message is to decode it, where is it using the blacklist to avoid parsing blocks from peers that have sent bad blocks in the past? Is the blacklist plugged into the connection manager, hence killing the connections right away?

Reply:
The blacklist is part of gossipsub, so i assume if i blacklist a peer, gossipsub will handle the rest and ensure i never get anything from them again